WordPress/gutenberg
# Try: Swap pencil icon for drawer icon in template details panel

## What?
In the template details panel there is an icon button (`pencil`) which invokes the full-screen editor on click. This PR replaces `pencil` with `drawerLeft` / `drawerRight` (according to `is_RTL()`).

## Why?
I've seen feedback that the pencil icon looks like a 'rename' affordance for the title. 

## Testing Instructions
1. In the site editor open navigate to a template e.g. Index.
2. Observe the pencil icon button is replaced with the drawer icon.

### Before
<img width="439" alt="Screenshot 2023-09-05 at 14 58 10" src="https://github.com/WordPress/gutenberg/assets/846565/cbab44e3-10ce-4212-9079-9d6ba059ea6d">


### After
<img width="427" alt="Screenshot 2023-09-05 at 14 57 53" src="https://github.com/WordPress/gutenberg/assets/846565/a77fc6c0-2af9-4e2a-91c9-bb580dde8e5f">

I'm not sure how successful this is so I've only updated the template details panel for now. Perhaps there are other solutions we might consider. What do y'all think?
